h5p_proplist.go

```diff
@@ -40,6 +44,26 @@ func (p *PropList) Close() error {
 	return p.closeWith(h5pclose)
 }
 
+// Sets the size of the chunks used to store a chunked layout dataset.
+// https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-SetChunk
+func (p *PropList) SetChunk(ndims uint, dim []uint) error {
+	var c_dim *C.hsize_t
+	c_dim = (*C.hsize_t)(unsafe.Pointer(&dim[0]))
+	if err := h5err(C.H5Pset_chunk(C.hid_t(p.id), C.int(ndims), c_dim)); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Sets deflate (GNU gzip) compression method and compression level.
+// https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-SetDeflate
+func (p *PropList) SetDeflate(level uint) error {
```

Comment:
perhaps we could make `level` an `int` instead, and piggy-back on the behaviour of `compress/flate` that uses `-1` to mean `DefaultCompression`?

Reply:
Maybe we should keep the behavior in C API to use uint.

Reply:
it's not really changing its behaviour: it's adding an additional possible value (-1) to mean something that isn't possible to express currently.

`cgo` based packages are a bit akward because one wants to follow the original C API (so switchers from C are in known territory) while at the same provide the Go idioms that made those switchers switch to Go.

anyways, I'll wait for @kortschak to break the tie :)

Reply:
This should be an `int`.

Reply:
I still stick to my opinion. `level` is between 0 and 9, and with no default value.

Reply:
Can you explain why it is that the C trumps Go idiom here?

Reply:
Sorry, I cannot find the rule for this.

Reply:
I don't understand. What rule?

Reply:
Can we communicate more efficiently? Just tell me your reason. 
Here, `level` is always a `uint`, and no need to define a default value. We need extra unnecessary check if use `int`. This is my point.

Reply:
There are a variety of reasons.

1. The compression routines already in use in the standard library use `int` types.
2. There already needs to be a check since any type in Go is able to hold more than the members of the set {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}.
3. The check is not expensive in comparison to all of the other work that is being done by any non-trivial program (and likely even any trivial program).
4. The default integer type is `int`.

I don't see any reason to keep this value as a `uint` except that the code here claims that C uses a `uint` (it doesn't because depending on the platform and compiler `uint` and `unsigned long long` may be different sizes - also note that that makes [this](https://github.com/gonum/hdf5/pull/48/files#diff-96202169867e203414c9ede99d46f5caR48) and [this](https://github.com/gonum/hdf5/pull/48/files#diff-96202169867e203414c9ede99d46f5caR53) unsafe).

The upshot is that hdf5 here is Go code, with Go coding conventions. It wraps C, but does that so that the users of this package do not need to interact directly with the C.

Reply:
The first one of your reasons is a good point. But the other three are unrelated. I already update it.

Reply:
No. They are very much related; in fact 4. is essentially a restatement of 1. Number 2. and 3. is a refutation of the importance of "We need extra unnecessary check if use `int`."